server/src/main/kotlin/delta/codecharacter/server/user/public_user/PublicUserService.kt

```diff
@@ -45,16 +45,35 @@ class PublicUserService(@Autowired private val publicUserRepository: PublicUserR
                 losses = 0,
                 ties = 0,
                 score = 0.0,
-                tier = LeaderBoardEnum.TIER_PRACTICE,
+                tier = TierTypeDto.TIER_PRACTICE,
                 isDailyChallengeCompleted = false,
                 tutorialLevel = 1,
             )
         publicUserRepository.save(publicUser)
     }
 
+    fun updateLeaderboard(publicUsers: List<PublicUserEntity>) {
+        for (i in publicUsers.indices) {
+            val user = publicUserRepository.findById(publicUsers[i].userId).get()
```

Comment:
- Don't do this. You have already queried the entire data and stored it in `publicUsers`. make use of it.
```suggestion
            val user = publicUsers[i]
```
- Improve `code-quality`